build.gradle

```diff
@@ -2,14 +2,18 @@ plugins {
     id 'scala'
     id 'checkstyle'
     id 'com.github.alisiikh.scalastyle' version '3.3.1'
-    id 'org.jetbrains.intellij' version '0.4.16'
-    id "org.sonarqube" version "2.8"
+    id 'org.jetbrains.intellij' version '0.4.21'
+    id 'org.sonarqube' version '3.0'
     id 'jacoco'
     id 'maven-publish'
 }
 
 group 'fi.aalto.cs'
-version '0.1.' + (System.getenv('TRAVIS_BUILD_NUMBER') ?: '0')
+version = System.getenv('TRAVIS_BUILD_NUMBER')
+        //  1379 is not a "magic number" it's # of Travis last build
+        //  on the moment of creating this code :kissing_heart:
+        ? ('1.0.' + (System.getenv('TRAVIS_BUILD_NUMBER') - 1379))
+        : '1.0.0'
```

Comment:
I suggest we just get rid of the build number here and go with standard `MAJOR.MINOR.PATCH` versioning. A single bug fix may increment the build number many times, but we actually only want to add +1 to the patch number with a single bugfix.

Reply:
Although now that I think about it, removing the build number probably would mean also removing the travis plugin publishing process

Reply:
I was going to suggest a versioning scheme `MAJOR.MINOR.PATCH+BUILD` like `1.0.3+319`, which is also a [standard](https://semver.org/#spec-item-10). (In **About** window, it could have been shown as **Version: 1.0.3 (build 319)**.) According to SemVer standard, two versions differing only by `+BUILD` part would be considered equal in precedence, which ideally would mean that IntelliJ _would not_ suggest an upgrade in case there's no increment in other parts.

Unfortunately, it seems that IntelliJ does not follow this standard (as far as I understand [this](https://github.com/JetBrains/intellij-community/blob/master/platform/util-rt/src/com/intellij/util/text/VersionComparatorUtil.java)), which means that the version upgrade is suggested also in case if only `+BUILD` number is incremented. That makes me think, perhaps, it's best to follow simple (manually assigned) `MAJOR.MINOR.PATCH` scheme (without `BUILD` number at all) because, as @nikke234 said, that would error publish stage of a test/master build if `PATCH` has been forgotten to change, which seems like a good thing. So I agree with @nikke234 here.

It would be possible to, nonetheless, store build number separately to be shown in the UI (in About window) if we are hyper-afraid of ending up in a situation where we don't know which version of our source code is behind some weird bug an user encounters. But as there should be guaranteed 1-to-0..1 mapping between build numbers and `MAJOR.MINOR.PATCH` strings (apart from test/master releases), that should be unnecessary.

One more thing: we could label `test` versions with some [pre-release identifier](https://semver.org/#spec-item-9) such as `1.0.4-test`. Is this something we would see useful? IntelliJ should handle that as expected (`1.0.4` < `1.0.5-test` < `1.0.5`).

Reply:
anything is fine as long as: a) IntelliJ IDEA will propose an update to users, b) we can unambiguously understand what is the contained code